L46-core-tls-credential-API.md

```diff
@@ -0,0 +1,339 @@
+   to the callback when it is invoked.
+   The callback can be set to null to disable callbacks.  Callers should generally
+   set it to null before they unref the distributor, to ensure that no subsequent
+   callbacks are invoked. */
+GRPCAPI void grpc_tls_certificate_distributor_set_watch_status_callback(
+    grpc_tls_certificate_distributor* distributor,
+    grpc_tls_certificate_watch_status_cb cb, void* user_data);
+```
+
+## Example Use
+
+We will provide an example showing how to put all these APIs together. Note that users wouldn't be able to interact directly with the core APIs.
+Each wrap language needs to wrap these APIs in a way that meets the inherent core logic.
+
+```c
+/* Create option and set basic security primitives. */
+grpc_tls_credentials_options* options = grpc_tls_credentials_options_create();
+grpc_tls_credentials_options_set_server_verification_option(options, GRPC_TLS_SKIP_HOSTNAME_VERIFICATION);
+grpc_tls_authorization_check_config* config = grpc_tls_authorization_check_config_create(.....);
+grpc_tls_credentials_options_set_authorization_check_config(options, config);
+grpc_tls_server_credentials_set_min_tls_version(options, TLS1_2);
+grpc_tls_server_credentials_set_max_tls_version(options, TLS1_3);
+grpc_tls_credentials_options_set_root_cert_name(options, "root_PEM_certs");
+grpc_tls_credentials_options_set_identity_cert_name(options, "identity_PEM_certs");
+/* Use the file-based provider for reloading certificates. */
+grpc_tls_certificate_provider* file_provider = grpc_tls_certificate_provider_file_watcher_create(...);
+grpc_tls_credentials_options_set_certificate_provider(options, file_provider);
+/* Create TLS channel credentials. */
+grpc_channel_credentials* creds = grpc_tls_credentials_create(options);
+```
+Wrap language should also allow users to define their own provider implementations:
+```c
+static grpc_tls_certificate_distributor* GetDistributor(grpc_tls_certificate_provider_external* external_provider) {
+    /* .... */
+}
+static void Destroy(grpc_tls_certificate_provider_external* external_provider) {
+    /* .... */
+    delete  external_provider;
+}
+grpc_tls_certificate_provider_external* external_provider = new grpc_tls_certificate_provider_external(); 
+external_provider->get_distributor = GetDistributor;
+external_provider->destroy = Destroy;
+/* The ownership will be taken by provider. */
+grpc_tls_certificate_provider* provider grpc_tls_certificate_provider_external_create(external_provider);
+
+```
+
+## Wrap Languages
+Each wrap language might have their own way to wrap these core APIs, but in general, there are a few things to consider:
+1. create an interface-like class `ProviderInterface` which contains two functions GetDistributor() and Destroy()
+2. create a concrete class `FileCertificateProvider` that implements `ProviderInterface` and wraps the provider created by `grpc_tls_certificate_provider_file_watcher_create`
```

Comment:
We need a concrete class `StaticCertificateProvider` that takes static certificates from memory, passed by application. So that we will be feature parity with `SslCredentialsOptions`.